File: src/main/webui/src/model/common/PropertyDefinition.ts
import Category from "./Category";

export default class PropertyDefinition {
    name: string;
    label: string;
    description: string;
    reverse: boolean;
    category: Category;
    unit: string;
    accessibility: string;
    isFree: boolean;
    decimal: number;
    constructor(name: string, label: string, description: string, reverse: boolean, category: Category, 
        unit: string, accessibility: string, isFree: boolean, decimal: number) {
        this.name = name;
        this.label = label;
        this.description = description;
        this.reverse = reverse;
        this.category = category;
        this.unit = unit;
        this.accessibility = accessibility;
        this.isFree = isFree;
        this.decimal = decimal;
    }
}

File: src/main/java/io/tbbc/cf/property/PropertyDefinition.java
package io.tbbc.cf.property;

import java.math.BigDecimal;

public record PropertyDefinition(PropertyName name, String label, String description, boolean reverse,
                                 CategoryName categoryName, BigDecimal minValue, BigDecimal maxValue, String unit,
                                 int decimal, Accessibility accessibility, boolean isFree) {
}
